smart-contract/src/contract.rs

```diff
@@ -2387,6 +2510,401 @@ mod tests {
+            _ => panic!("unexpected execute error"),
+        }
+
+        // Try to remove executor by user not admin
+        let err = execute(
+            deps.as_mut(),
+            mock_env(),
+            mock_info("notadmin", &[]),
+            ExecuteMsg::RemoveExecutor {
+                id: "executor2".into(),
+            },
+        )
+        .unwrap_err();
+
+        // Ensure the expected error was returned.
+        match err {
+            ContractError::Unauthorized {} => {}
+            _ => panic!("unexpected execute error"),
+        }
+
+        // Try to remove executor does not exist.
+        let err = execute(
+            deps.as_mut(),
+            mock_env(),
+            mock_info("admin", &[]),
+            ExecuteMsg::RemoveExecutor {
+                id: "executor".into(),
+            },
+        )
+        .unwrap_err();
+
+        // Ensure the expected error was returned.
+        match err {
+            ContractError::Std(StdError::GenericErr { msg, .. }) => {
+                assert_eq!(msg, "executor does not exist")
+            }
+            _ => panic!("unexpected execute error"),
+        }
+    }
+
+    #[test]
+    fn executor_transfer_test() {
+        // Create mock deps.
+        let mut deps = mock_dependencies(&[]);
+
+        // Init
+        instantiate(
+            deps.as_mut(),
+            mock_env(),
+            mock_info("admin", &[]),
+            InitMsg {
+                denom: "dcc.coin".into(),
+            },
+        )
+        .unwrap();
+
+        // Create join member
+        execute(
+            deps.as_mut(),
+            mock_env(),
+            mock_info("admin", &[]),
+            ExecuteMsg::Join {
+                id: "bank".into(),
+                name: "bank".into(),
+                kyc_attrs: vec!["bank.kyc.pb".into()],
+            },
+        )
+        .unwrap();
+
+        // Add executor
+        execute(
+            deps.as_mut(),
+            mock_env(),
+            mock_info("admin", &[]),
+            ExecuteMsg::AddExecutor {
+                id: "executor".into(),
+            },
+        )
+        .unwrap();
+
+        // Assume the customer has a balance of tokens + the required attribute.
+        let dcc = coin(1000, "dcc.coin");
+        deps.querier
+            .base
+            .update_balance(Addr::unchecked("customer"), vec![dcc]);
+        deps.querier
+            .with_attributes("customer", &[("bank.kyc.pb", "ok", "string")]);
+
+        // Transfer dcc from the customer to a member bank.
+        let res = execute(
+            deps.as_mut(),
+            mock_env(),
+            mock_info("executor", &[]),
+            ExecuteMsg::ExecutorTransfer {
+                amount: Uint128::new(500),
+                sender: "customer".into(),
+                recipient: "bank".into(),
+            },
+        )
+        .unwrap();
+
+        // Ensure message was created.
+        // TODO: validate marker transfer message...
```

Comment:
TODO?

Reply:
yeah this test is actually copied from the other transfer test; its still a todo there